### PR TITLE
Emit partial

### PR DESCRIPTION
**Pull request by juanjoDiaz**
Supersedes #36.
Also solves the issue raised in #31.

@lunaris this is heavily influenced by your PR.
Sorry it took me so long 😅
Wdyt? Does this cover your use case?

@chrisciszak this is a proper solution to you request so you don't have to rely on the library internals.
Wdyt?

If the feedback is good, I'll merge and release this.

### CODE REVIEW

File: packages/node/README.md [hunk 1]
```diff
@@ -157,6 +158,28 @@ Imagine an endpoint that send a large amount of JSON objects one after the other
...
+Imagine an endpoint that send a large amount of JSON objects one after the other (`"Once upon a midnight <...>"`).
+
+```js
+  import { JSONParser } from '@streamparser/json-node';
+
+  const parser = new JSONParser({ stringBufferSize: undefined, paths: ['$.*'], keepStack: false });
+
+  const response = await fetch('http://example.com/');
+
+  const reader = response.body.pipe(parse)getReader();
```
Comment (brandonros) [inline packages/node/README.md:172]: typo here `.getReader`

File: packages/node/dist/deno/README.md [hunk 2]
```diff
@@ -157,6 +158,28 @@ Imagine an endpoint that send a large amount of JSON objects one after the other
...
+Imagine an endpoint that send a large amount of JSON objects one after the other (`"Once upon a midnight <...>"`).
+
+```js
+  import { JSONParser } from '@streamparser/json-node';
+
+  const parser = new JSONParser({ stringBufferSize: undefined, paths: ['$.*'], keepStack: false });
+
+  const response = await fetch('http://example.com/');
+
+  const reader = response.body.pipe(parse)getReader();
```
Comment (brandonros) [inline packages/node/dist/deno/README.md:172]: typo here .getReader

File: packages/node/README.md [hunk 3]
```diff
@@ -157,6 +158,28 @@ Imagine an endpoint that send a large amount of JSON objects one after the other
...
+
+  reader.on('data', ({ value, key, parent, stack }) => {
+    if (partial) {
```
Comment (lunaris) [inline packages/node/README.md:175]: Is `partial` defined here?

File: packages/node/dist/deno/README.md [hunk 4]
```diff
@@ -157,6 +158,28 @@ Imagine an endpoint that send a large amount of JSON objects one after the other
...
+
+  reader.on('data', ({ value, key, parent, stack }) => {
+    if (partial) {
```
Comment (lunaris) [inline packages/node/dist/deno/README.md:175]: Same as before -- is `partial` defined?

**Comment by lunaris** [comment]
Thanks for following up. I've given it a look through and I think it seems about the same, though I appreciate some of it has been changed to accommodate the existing design.

**Comment by juanjoDiaz** [comment]
Thanks for the review guys!

@lunaris it is indeed roughly the same. Except that, instead of having separate tokenizer, token-parser & json-parser, I just made the normal ones configurable. And extend the Node and WHATWG interfaces to also support this.

If there are not more comments, I'll merge and cut a release soon :)

**Pull request merged**